pkg/scaffold/internal/filesystem/errors.go

```diff
@@ -78,13 +96,19 @@ type createFileError struct {
 	err  error
 }
 
+// Error implements error interface
 func (e createFileError) Error() string {
 	return fmt.Sprintf("failed to create %s: %v", e.path, e.err)
 }
 
+// Unwrap implements Wrapper interface
+func (e createFileError) Unwrap() error {
+	return e.err
+}
+
 // IsCreateFileError checks if the returned error is because the file could not be created
-func IsCreateFileError(e error) bool {
-	_, ok := e.(createFileError)
+func IsCreateFileError(err error) bool {
+	_, ok := err.(createFileError)
```

Comment:
prob should move to `return errors.As(err, createFileError{})`

Reply:
I feel like that is less descriptive. IMO it seems to try to use a new feature when it doesn't really fit, I may just not be used to it.

Reply:
The thing is that they do not do exactly the same. Right now it only detects if the error returned is of that type, and using `errors.As(err, ERROR_TYPE)` will check if any of the chained errors is of that type. Do we want to move towards that?

Reply:
I say we should move towards using `errors.As` since the underlying error may have been wrapped by an anonymous `fmt.Errorf` for a more informative error message by a caller.

Reply:
Done